fivekmrun_app_flutter/lib/state/locale_provider.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'dart:io';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class LocaleProvider extends ChangeNotifier {
+  Locale? _locale;
+
+  Locale? get locale => _locale;
+
+  LocaleProvider() {
+    _loadLocale();
+  }
+
+  Future<void> setLocale(Locale locale) async {
+    final preferences = await SharedPreferences.getInstance();
+    await preferences.setString('locale', locale.languageCode);
+    _locale = locale;
+    notifyListeners();
+  }
+
+  Future<void> _loadLocale() async {
+    final preferences = await SharedPreferences.getInstance();
+    final localeCode = preferences.getString('locale');
+
+    if (localeCode != null) {
+      _locale = Locale(localeCode);
+    } else {
+      Locale platformLocale = Locale(Platform.localeName);
+      _locale = (AppLocalizations.supportedLocales.contains(platformLocale))
+          ? platformLocale
+          : Locale("en");
```

Comment:
WDYT of leaving `bg` as the default language?
```suggestion
          : Locale("bg");
```

Reply:
Yeah, it makes sense. We are targeting only Bulgarians and the app doesn't make sense for foreigners. Let's leave BG as the default.

Reply:
My thought process for having "en" as fallback was the following: if the device is neither in BG nor EN, chances are it belongs to somebody who does not speak Bulgarian. And having the translation makes the app available to foreigners living in Bulgaria.